src/curl-stream.js

```diff
@@ -0,0 +1,49 @@
+let Readable = require('stream').Readable;
+let Curl = require('node-libcurl').Curl;
+
+// https://github.com/JCMais/node-libcurl/blob/develop/examples/progress-callback.js
+class CurlStream extends Readable {
+  constructor(url, opts) {
+    super(opts);
+    let c = this.curl = new Curl();
+    c.setOpt('URL', url);
+    c.setOpt(Curl.option.NOPROGRESS, false);
+    // NOT SECURE!!! I should instead do HEAD on the URL
+    // and see if the Location: header is set to the same
+    // protocol as the original request
+    // see http://curl.haxx.se/mail/lib-2007-10/0025.html
+    c.setOpt(Curl.option.FOLLOWLOCATION, true); 
```

Comment:
We should use:
http://curl.haxx.se/libcurl/c/CURLOPT_REDIR_PROTOCOLS.html

To only allow follow `https`... In particular I'm a tiny bit scared that someone could redirect to FTP and LDAP (whatever the LDAP protocol is). It seems curl forbids most of the worst by default:
http://curl.haxx.se/libcurl/c/CURLOPT_FOLLOWLOCATION.html

But people shouldn't do HTTP, and worst case it could be used to steal something that's always on a special ip, like AWS metadata service, which could contain credentials... Note: we might not need redirects at all.. Pretty sure S3 never gives us a redirect when we use the bucket domain name option for access (something like`<bucket>.<region>....amazonaws.com`).